types/Promise.d.ts

```diff
@@ -4,6 +4,41 @@
 // Based on roblox-lua-promise v3.1.0
 // https://eryn.io/roblox-lua-promise/
 
+interface PromiseErrorOptions {
+	trace: string;
+	context: string;
+	kind: keyof PromiseErrorConstructor["Kind"];
```

Comment:
I think we could make this `kind: Promise.Error.Kind;` with a bit of namespace wrangling?